File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    let canSay = ["aya", "ye", "woo", "ma"];
    var answer = 0;
    
    for(let i = 0; i < babbling.length; i++){
        let babble = babbling[i];
        
        for(let j = 0; j < canSay.length; j++){
            // 연속 2번 나오면 break
            if(babble.includes(canSay[j].repeat(2))){
                break;
            }
            babble = babble.split(canSay[j]).join(" ");
        }
        
        if(babble.split(" ").join("").length === 0){
            answer += 1;
        }
    }
    return answer;
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
class Solution {
    public int solution(String[] babbling) {
        String[] canSay = {"aya", "ye", "woo", "ma"};
        int answer = 0;
        
        for(int i = 0; i < babbling.length; i++) {
            String babble = babbling[i];
            
            for(int j = 0; j < canSay.length; j++) {
                // 연속 2번 나오면 break
                if(babble.contains(canSay[j] + canSay[j])) {
                    break;
                }
                babble = babble.replace(canSay[j], " ");
            }
            
            if(babble.replaceAll(" ", "").length() == 0) {
                answer += 1;
            }
        }
        return answer;
    }
}